src/tests/test_interaction.cc

```diff
@@ -24,82 +24,84 @@
 #include "../../include/votca/csg/interaction.h"
 #include "../../include/votca/csg/molecule.h"
 #include "../../include/votca/csg/topology.h"
+
 #include <string>
+#include <votca/tools/constants.h>
 
 using namespace std;
 using namespace votca::csg;
+using namespace votca::tools;
 
 BOOST_AUTO_TEST_SUITE(interaction_test)
 
 BOOST_AUTO_TEST_CASE(test_interaction_constructor) {
-  IBond bond1(1, 2);
-  IAngle angle1(1, 2, 3);
-  IDihedral dihedral(1, 2, 3, 4);
+  Topology top;
+  top.CreateInteraction(std::list<votca::Index>{1, 2}, "grp_name", 0,
+                        topology_constants::unassigned_molecule_id);
+  top.CreateInteraction(std::list<votca::Index>{1, 2, 3}, "grp_name", 1,
+                        topology_constants::unassigned_molecule_id);
+  top.CreateInteraction(std::list<votca::Index>{1, 2, 3, 4}, "grp_name", 2,
+                        topology_constants::unassigned_molecule_id);
 }
 
 BOOST_AUTO_TEST_CASE(test_interaction_setters_getters) {
 
-  IBond bond1(1, 2);
-  bond1.setGroup("large");
-  bond1.setGroupId(1);
-  bond1.setIndex(1);
-  bond1.setMolecule(1);
+  Topology top;
+  const Interaction* bond1 =
+      top.CreateInteraction(std::list<votca::Index>{1, 2}, "large", 1, 1);
```

Comment:
This is what I consider  a bad idea. Why can I not create Interactions without a topology?

Reply:
No, it does not make it more flexible, it reduces scope. In this case flexibility is not needed. If you look into the topology object CreateInteraction method you can get some insight as to why you would not want to change the interaction outside of the topology.